server/metadata/key_encoder.go

```diff
@@ -63,6 +65,18 @@ func (d *DictKeyEncoder) EncodeIndexName(idx *schema.Index) []byte {
 	return d.encodedIdxName(idx)
 }
 
+func (d *DictKeyEncoder) EncodeKeyWithPartition(encodedTable []byte, idx *schema.Index, idxValue interface{}, partition uint16) (keys.Key, error) {
+	var allParts []interface{}
+	allParts = append(allParts, encoding.UInt16ToByte(partition))
+	allParts = append(allParts, idxValue)
+	return d.EncodeKey(encodedTable, idx, allParts)
+}
+
+func (d *DictKeyEncoder) DecodeKeyWithPartition(key keys.Key) (interface{}, uint16) {
```

Comment:
Can you also document this API that this is returning index value and partition?